[SYCL][Doc] Require put link to test PR in commit message (#3010)
intel/llvm: CONTRIBUTING.md

- For any DPC++-related commit, the `[SYCL]` tag should be present in the
  commit message title. To a reasonable extent, additional tags can be used
  to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.

> - For product changes which require modification of E2E tests

Reviewer: It probably makes sense to replace E2E here with more generic: "tests, outside of this repository"

Author: done

  (see [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain))

>   the commit message should contain link to corresponding test PR, e.g.: E2E

Reviewer 2: Could you please clarify if the intent here is that a link should be in final commit message or in the first PR message or both?

Reviewer: Thinking more about that... Should we also require that PR to have clear message which will be used as final merge commit message when many commits squashed?
If we do this link to test should be in PR.

Author: done

- For any DPC++-related commit, the `[SYCL]` tag should be present in the
  commit message title. To a reasonable extent, additional tags can be used
  to signify the component changed, e.g.: `[PI]`, `[CUDA]`, `[Doc]`.

> - For product changes which require modification in tests outside current repo

Reviewer 3: ```suggestion
- For product changes which require modification in tests outside of the current repository
```
?

>   (see [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain))

Reviewer 3: ```suggestion
  (see [Test DPC++ toolchain](sycl/doc/GetStartedGuide.md#test-dpc-toolchain)),
```

>   the commit message should contain link to corresponding test PR, e.g.: E2E

Reviewer 3: ```suggestion
  the commit message should contain the link to corresponding test PR, e.g.: E2E
```

  test changes are available under intel/llvm-test-suite#88 or SYCL
  conformance test changes are available under KhronosGroup/SYCL-CTS#65 (see
  [Autolinked references and URLs](https://docs.github.com/en/free-pro-team/github/writing-on-github/autolinked-references-and-urls)
  for more details). The same message should be present both in commit

>   message and PR description.

Reviewer 3: ```suggestion
  message and in PR description.
```
